backend_uml.go

```diff
@@ -62,23 +63,63 @@ func (b umlBackend) KernelPath() (string, error) {
 
 func (b umlBackend) ModulePath() (string, error) {
 	// make sure the UML modules exist
-	// on non-merged usr systems the modules still reside under /usr/lib/uml
+
+	/* On debian systems, the module reside under the first child directory under
+	 * /usr/lib/uml/modules; on non-merged usr systems the modules still reside
+	 * under /usr/lib/uml
+	 */
 	moddir := "/usr/lib/uml/modules"
-	if _, err := os.Stat(moddir); err != nil {
-		return "", fmt.Errorf("user-mode-linux modules not installed")
-	}
+	if _, err := os.Stat(moddir); err == nil {
+		// find the subdirectory containing the modules for the UML release
+		modSubdirs, err := ioutil.ReadDir(moddir)
+		if err != nil {
+			return "", err
+		}
 
-	// find the subdirectory containing the modules for the UML release
-	modSubdirs, err := ioutil.ReadDir(moddir)
-	if err != nil {
-		return "", err
+		if len(modSubdirs) != 1 {
+			return "", fmt.Errorf("could not determine which user-mode-linux modules to use")
+		}
+
+		moddir = path.Join(moddir, modSubdirs[0].Name())
+		if _, err := os.Stat(moddir); err != nil {
+			return "", err
+		}
+
+		return moddir, nil
 	}
-	if len(modSubdirs) != 1 {
-		return "", fmt.Errorf("could not determine which user-mode-linux modules to use")
+
+	/* On Arch systems, however, the modules are stored under /usr/lib/modules/
+	 * with a suffix, "-usermodelinux".
+	 */
```

Comment:
Out of curiosity: what does Fedora do about the modules? Looking around shows absolutely nothing.